### PR TITLE
Setup dagger2

### PR DESCRIPTION
**Pull request by jimzhaozhengwang**
setup Dagger 2

https://dagger.dev/

### CODE REVIEW

File: android/app/src/main/java/com/chorestory/module/NetModule.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import com.google.gson.GsonBuilder;
+
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+
+@Module
```
Comment (keller00) [inline android/app/src/main/java/com/chorestory/module/NetModule.java]: I'm gonna look up some tutorials, but I would prefer if we could setup a meeting where you explained Dagger a bit more interactively.

File: android/app/src/main/java/com/chorestory/app/App.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+public class App extends Application {
+    private static AppComponent appComponent;
+
+    public static AppComponent getAppComponent() {
+        return appComponent;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
```
Comment (keller00) [inline android/app/src/main/java/com/chorestory/app/App.java]: Make sure @cr2david knows about these, as he has done this too.

File: android/.idea/inspectionProfiles/Project_Default.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
+  <profile version="1.0">
+    <option name="myName" value="Project Default" />
+    <inspection_tool class="AndroidLintNewerVersionAvailable" enabled="true" level="WARNING" enabled_by_default="true" />
+  </profile>
+</component>
```
Comment (keller00) [inline android/.idea/inspectionProfiles/Project_Default.xml]: Missing newline

Comment (keller00) [reply android/.idea/inspectionProfiles/Project_Default.xml]: Just fix this before merging

File: android/app/src/main/java/com/chorestory/module/NetModule.java (hunk 1 above)
Comment (cr2david) [reply android/app/src/main/java/com/chorestory/module/NetModule.java]: big agree

**Pull request merged**